Sources/VitaminUIKit/Components/Snackbar/VitaminSnackbar.swift

```diff
@@ -0,0 +1,360 @@
+        withImageConstraints.append(
+            messageLabel.rightAnchor.constraint(equalTo: self.rightAnchor, constant: -rightMargin)
+        )
+    }
+
+    // set up layout constraints for no image case without activating them
+    private func setupWithoutImageConstraints() {
+        withoutImageConstraints.append(titleLabel.centerXAnchor.constraint(equalTo: self.centerXAnchor))
+        withoutImageConstraints.append(messageLabel.centerXAnchor.constraint(equalTo: self.centerXAnchor))
+        withoutImageConstraints.append(
+            messageLabel.widthAnchor.constraint(lessThanOrEqualTo: self.widthAnchor, constant: -2 * rightMargin)
+        )
+    }
+
+    // apply changes when title or message are changed
+    private func applyNewTexts() {
+        titleLabel.attributedText = title.styled(as: .footnoteBold, with: VitaminColor.Core.Content.primary)
+        messageLabel.attributedText = message.styled(as: .footnote, with: VitaminColor.Core.Content.tertiary)
+    }
+
+    // apply changes when an image is set or changed
+    private func applyNewImage() {
+        if let image = image {
+            icon.image = image
+            if self.subviews.first(where: { $0 == icon }) == nil {
+                self.addSubview(icon)
+            }
+            NSLayoutConstraint.deactivate(withoutImageConstraints)
+            NSLayoutConstraint.activate(withImageConstraints)
+            titleLabel.textAlignment = .left
+            messageLabel.textAlignment = .left
+            let margins = rightMargin + imageWidthAndHeight + imageLeftAndRightMargin
+            messageLabel.preferredMaxLayoutWidth = maxWidth - margins
+        } else {
+            NSLayoutConstraint.activate(withoutImageConstraints)
+            NSLayoutConstraint.deactivate(withImageConstraints)
+            icon.removeFromSuperview()
+            titleLabel.textAlignment = .center
+            messageLabel.textAlignment = .center
+            messageLabel.preferredMaxLayoutWidth = maxWidth - 2 * rightMargin
+        }
+    }
+
+    // launch or invalidate timer for aut dismissal when the autodismiss prperty is set
+    private func applyNewAutoDismiss(_ newValue: Bool) {
+        if autoDismiss, !newValue {
+            timer?.invalidate()
+        } else if !autoDismiss, newValue {
+            launchAutoDismissal()
+        }
+    }
+
+    // apply changes when dismiss delay is set
+    private func applyNewDismissDelay() {
+        timer?.invalidate()
+        if autoDismiss {
+            launchAutoDismissal()
+        }
+    }
+
+    // apply changes when dismissOnTap property is set
+    private func applyNewDismissOnTap() {
+        if let gestureRecognizer = gestureRecognizer {
+            if dismissOnTap {
+                self.addGestureRecognizer(gestureRecognizer)
+            } else {
+                self.removeGestureRecognizer(gestureRecognizer)
+            }
+        }
```

Comment:
It could be nice to init `gestureRecognizer` only when needed.

Reply:
done